code/routers/user/donation.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express')
+const router = express.Router();
+
+const Donation = require('../../models/donation');
+
+const donationCodes = require('../../config/resCodes').donation;
+
+/* 
+	Donation API:
+		Create new donation:
+			- Input: donation info, newTripID, and a callback function
+			- Create new mongoose donation object with given info
+			- Save to database
+			- Call callback with either success or fail information
+
+	A Donation follows the following schema:
+		itemName: {type: String, required: true},
+	    rating: {type: Number, required: true},
+	    locationID: {type: LocationSchema, required: true},
+	    tripID: {type:String, required:true},
+	    Category: {type:String, enum:['Health', 'Household', 'Clothing', 'Education', 'Art', 'Sports', 'Miscellaneous', 'Animal Welfare', 'Environment', 'Food']},  
+	    donationDateTime: Date,
+	    reportingDateTime: {type: Date, default: Date.now},
+	    itemDescr: String,
+	    pictures: String, //String for now, maybe use gridfs?
+	    Organization: Boolean, //could there be a better way to do this?
+	    Suggestion: Boolean //could there be a better way to do this?
```

Comment:
It is good to be thorough, but this does not need to be included. If someone needs to know about the structure of the donation schema they can just look at the schema or the diagram.

Reply:
Okay sure I'll get rid of it.